### DISCUSSION TITLE
Initialize provider from command line argument

### DISCUSSION THREAD

**Discussion by mstagitis** [rrousselGit/riverpod#3088] [Q&A]
Hi all - When I run my Flutter app, I pass in a handful of dependency server addresses via command line argument. Ideally, I'd like to stash these in a Provider (that will never change) to make them easily available to the rest of my app's state. Is this possible? If not, what's the best practice in this situation?

Thank you!
-- Matt

**Comment by mstagitis** [comment]
As I think about this - one basic solution is to initialize the provider states to `null` and then immediately update the states with the real addresses in the root widget (when they become known). This approach should work, but feels hacky to me... Is there a better way?

**Comment by AhmedLSayed9** [comment]
You can just make your provider return the environment/configuration variable:
```dart
@Riverpod(keepAlive: true)
String serverAddress(ServerAddressRef ref) {
  return String.fromEnvironment('server_address');
}
```
